### PR TITLE
format: allow input from stdin

### LINKED ISSUE #13791: Accept stdin as argument to `format`
**Feature request**
At the moment meson format only accepts files, but it would be useful to also accept stdin in some way (muon does this with "-" as argument)

This would be especially useful for ides and extension such as https://github.com/mesonbuild/vscode-meson

Also see https://github.com/mesonbuild/vscode-meson/pull/261

### PR DESCRIPTION
**Pull request by bruchar1**
Fixes #13791

### CODE REVIEW

File: mesonbuild/mformat.py [hunk 1]
```diff
@@ -979,7 +986,10 @@ def run(options: argparse.Namespace) -> int:
             src_file = src_file / build_filename
 
         try:
-            code = src_file.read_text(encoding='utf-8')
+            if from_stdin:
+                code = sys.stdin.read()
```
Comment (dcbaker) [inline mesonbuild/mformat.py:990]: I think the error message looks funny for the stdin case.

**Review by dcbaker — CHANGES_REQUESTED** [review]
One small nit about error messages.

How does this work with finding `meson.format`, since that's based on paths?

**Comment by bruchar1** [comment]
> How does this work with finding `meson.format`, since that's based on paths?

Parent path of `Path('-')` will be current working directory. Therefore, it will look for `meson.format` in the current working directory. Does it makes sense?

File: mesonbuild/mformat.py (hunk 1 above)
Comment (bruchar1) [reply mesonbuild/mformat.py]: Good point! :+1:

**Pull request merged**